x-pack/plugins/apm/public/components/app/service_overview/service_overview_instances_chart_and_table.tsx

```diff
@@ -95,6 +95,10 @@ export function ServiceOverviewInstancesChartAndTable({
     status: mainStatsStatus,
   } = useFetcher(
     (callApmApi) => {
+      if (!transactionType && transactionTypeStatus === FETCH_STATUS.SUCCESS) {
+        return Promise.resolve(undefined);
+      }
+
       if (!start || !end || !transactionType || !latencyAggregationType) {
         return;
       }
```

Comment:
It feels fragile that we take advantage of the fact that there's a difference between `Promise.resolve(undefined)` and `undefined`. 

But I don't want to block this PR or require any bigger changes so let's just keep what you have. 

In the future, it would be great if this was more explicit (pseudo):

```ts
      if (!transactionType && transactionTypeStatus === FETCH_STATUS.SUCCESS) {
        return FETCH_STATUS.ABORT;
      }

      if (!start || !end || !transactionType || !latencyAggregationType) {
        return FETCH_STATUS.PENDING;
      }
```